lib/elixir/lib/exception.ex

```diff
@@ -636,9 +636,9 @@ defmodule UndefinedFunctionError do
       " is undefined (module #{inspect module} is not available)"
   end
 
-  def message(%{reason: :"function not exported",  module: module, function: function, arity: arity}) do
+  def message(%{reason: :"function not exported",  module: module, function: function, arity: arity} = e) do
```

Comment:
Let's match on exports on the function head since we do it for all other parameters. :D